webapp/packages/plugin-sql-editor/src/SqlResultTabs/SqlResultTabsService.ts

```diff
@@ -75,16 +77,16 @@ export class SqlResultTabsService {
     }
   }
 
-  async closeTabGroup(state: ISqlEditorTabState, tabId: string) {
+  async closeTabGroup(state: ISqlEditorTabState, tabsContext: ITabsContext, context: IDataContextProvider, tabId: string) {
     const resultTab = state.resultTabs.find(tabState => tabState.tabId === tabId);
     const groupResultTabs = state.resultTabs.filter(tab => tab.groupId === resultTab?.groupId);
 
     if (groupResultTabs.length) {
       for (const groupTab of groupResultTabs) {
-        const canClose = await this.sqlQueryResultService.canCloseResultTab(state, groupTab.tabId);
+        const canClose = tabsContext.canClose(groupTab.tabId);
 
         if (canClose) {
-          this.removeResultTab(state, groupTab.tabId);
+          tabsContext.close(groupTab.tabId);
```

Comment:
you don't need to check canClose manually 
`tabsContext.close(groupTab.tabId);` will do it for you
<img width="377" alt="Screenshot 2024-03-29 at 7 08 06 PM" src="https://github.com/dbeaver/cloudbeaver/assets/811729/27a686d3-9fcd-4bc6-928b-4cb39a7c8c4a">